src/wp-includes/class-wp-term.php

```diff
@@ -103,17 +103,19 @@ final class WP_Term {
 	 * Retrieve WP_Term instance.
 	 *
 	 * @since 4.4.0
+	 * @since 6.4.0 Added the `$filter` parameter.
 	 *
 	 * @global wpdb $wpdb WordPress database abstraction object.
 	 *
-	 * @param int    $term_id  Term ID.
-	 * @param string $taxonomy Optional. Limit matched terms to those matching `$taxonomy`. Only used for
-	 *                         disambiguating potentially shared terms.
+	 * @param int          $term_id  Term ID.
+	 * @param string       $taxonomy Optional. Limit matched terms to those matching `$taxonomy`. Only used for
+	 *                               disambiguating potentially shared terms.
+	 * @param string       $filter Optional. How to sanitize term fields. Default 'raw'.
 	 * @return WP_Term|WP_Error|false Term object, if found. WP_Error if `$term_id` is shared between taxonomies and
 	 *                                there's insufficient data to distinguish which term is intended.
 	 *                                False for other failures.
 	 */
-	public static function get_instance( $term_id, $taxonomy = null ) {
+	public static function get_instance( $term_id, $taxonomy = null, $filter = 'raw' ) {
```

Comment:
I'm a bit hesitant about making `raw` the default filter that is applied to existing code calling the `WP_Term::get_instance()` method. This could potentially lead to situations where existing code calling this method fails to have any filter applied, where previously it did (an edge case, for sure). What if instead, the default is null and we update the places in `get_term()` that calls this method to intentionally pass 'raw' instead, knowing that the correct filter value will be applied in `get_term()` after the instance is created?